superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/AGGridViz.tsx

```diff
@@ -432,6 +434,32 @@ export default function AGGridViz({
         />,
       ];
     }
+    if (
+      assemblyLineUrl &&
+      selectedData.typeData.eml_path &&
+      selectedData.typeData.eml_path.length > 0
+    ) {
+      specialMenuItems = [
+        ...specialMenuItems,
+        <SubmitToAssemblyLineMenuItem
+          onSelection={handleContextMenu}
+          label={
+            selectedData.typeData.eml_path.length > 1
+              ? 'Submit files to ASSEMBLYLINE'
+              : 'Submit file to ASSEMBLYLINE'
+          }
+          key="submit-file-to-assembly-line"
+          data={selectedData.typeData.eml_path}
+          base_url={assemblyLineUrl}
+          disabled={selectedData.typeData.eml_path.length > SUBMISSION_LIMIT}
```

Comment:
here you are checking the length of the entire list, but on line 456 you are checking for the number of unique items. I think it should be consistent for the rare case of duplicate values. You are also passing the entire list to the component in data on line 452.

Reply:
I am passing the entire list since this allows for multiple submissions to be made to AssemblyLine at the same time, part of the multi select functionality

Reply:
I just noticed that retain EML records to Alfred' has a similar structure:

 if (
      enableAlfred &&
      selectedData.typeData.harmonized_email_id &&
      selectedData.typeData.harmonized_email_id.length > 0
    ) {
      specialMenuItems = [
        ...specialMenuItems,
        <RetainEmlMenuItem
          onSelection={handleContextMenu}
          label={
            selectedData.typeData.harmonized_email_id.length > 1
              ? 'Retain EML records to ALFRED'
              : 'Retain EML record to ALFRED'
          }
          key="retain-eml"
          data={{
            email_ids: Array.from(
              new Set(selectedData.typeData.harmonized_email_id),
            ),
            dates: [
              ...Object.entries(selectedData.typeData)
                .filter(k => k[0].includes('TIME'))
                .map(v => v[1])
                .flat(),
              ...(selectedData.typeData.DATE || []),
              ...(selectedData.typeData.DATETIME || []),
            ],
          }}
          disabled={
            selectedData.typeData.harmonized_email_id.length > RETENTION_LIMIT
          }
          tooltip={
            new Set(selectedData.typeData.harmonized_email_id).size >
            RETENTION_LIMIT
              ? `Cannot retain more than ${RETENTION_LIMIT} unique harmonized email IDs.`
              : undefined
          }
        />,
      ];
    }
    
    Should we change this too to only consider unique harmonized email IDs?

Reply:
Good catch! i think the alfred menu is wrong too in that case they should both check unique number of items selected, and not just all selected ids